docs/en/observability/apm/configure/outputs/logstash.asciidoc

```diff
@@ -129,23 +129,40 @@ In addition to `@metadata`, APM Server provides other potentially useful fields,
 `data_stream` field, which can be used to conditionally operate on
 {observability-guide}/apm-data-model.html[event types], namespaces, or datasets.
 
-As an example, you might want to use {ls} to route all `metric` events to the same custom metrics data stream,
-rather than to service-specific data streams:
+As an example, you might want to use {ls} to route all `metrics` events to the same custom metrics data stream,
+rather than to service-specific data streams.
+
+However, if when you combine all `metrics` events there are events that have the `data_stream.dataset` field
+set to different values, indexing will fail with a message stating that the field does not accept any other values.
+This is because the `data_stream.dataset` field's mapping is set to `constant_keyword`,
+which expects all values of the fields in the index to be the same.
+
+To prevent losing data due to failed indexing, you should either:
+
+* Create a `@custom` component template to override the fields' mapping from `constant_keyword` to `keyword`.
+Read more in <<apm-custom-index-template>>.
+* Add a {logstash-ref}/plugins-filters-mutate.html[Logstash mutate filter] to update the value of `data_stream.dataset`.
+This option is shown in the example below.
```

Comment:
@marclop do you think we should keep creating a custom component template as an option or only suggest the latter?

Reply:
Let's just keep the latter. Although you _can_ override the type. It's not necessary to change the type when automatically routing based on the `data_stream.dataset` field value.